api/v1alpha2/azurecluster_conversion.go

```diff
@@ -63,6 +66,17 @@ func (src *AzureCluster) ConvertTo(dstRaw conversion.Hub) error { // nolint
 			for _, dstSubnet := range dst.Spec.NetworkSpec.Subnets {
 				if dstSubnet != nil && dstSubnet.Name == restoredSubnet.Name {
 					dstSubnet.RouteTable = restoredSubnet.RouteTable
+
+					for _, restoredIngress := range restoredSubnet.SecurityGroup.IngressRules {
+						if restoredIngress != nil {
+							for _, dstIngress := range dstSubnet.SecurityGroup.IngressRules {
+								if dstIngress != nil && dstIngress.Description == restoredIngress.Description {
```

Comment:
If Description isn't a unique field it's won't be reliable to use it to match restored ingress rules with existing ones since there might be two rules with the same description

Reply:
Subnet name above is unique (and that's enforced both by Azure and by CAPZ in validating webhooks)